lib/widgets/login.dart

```diff
@@ -327,24 +327,32 @@ class _EmailPasswordLoginPageState extends State<EmailPasswordLoginPage> {
   @override
   Widget build(BuildContext context) {
     assert(!PerAccountStoreWidget.debugExistsOf(context));
-
-    final emailField = TextFormField(
-      key: _emailKey,
-      autofillHints: const [AutofillHints.email],
+    final requireEmailFormatUsernames = widget.serverSettings.requireEmailFormatUsernames;
+
+    final usernameField = TextFormField(
+      key: _usernameKey,
+      autofillHints: [
+        AutofillHints.email,
+        if (!requireEmailFormatUsernames) AutofillHints.username,
```

Comment:
Hmm, interesting that this can be more than one!

Ah, but there is this caveat:
https://api.flutter.dev/flutter/material/TextField/autofillHints.html
> On iOS, only the first hint in the list is accounted for.

So probably we want `username` to come before `email` when `!requireEmailFormatUsernames`, i.e. when we're including `username` at all.

(That caveat should probably be more prominent in the docs, like up in that intro section that ends with "Defaults to an empty list." As is, it's kind of buried in a lot of text.)